src/core/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,184 @@
+  className?: string;
+  /** Is there any need to have as prop */
+  asProp?: asPropType;
+  /** aria-label for pagination  */
+  'aria-label': string;
+  /** Function for page number indicator text */
+  textFunction?: (currentPage: number, lastPage: number) => string;
+  /** Use small screen styling */
+  smallScreen?: boolean;
+  /** Number of the current page */
+  currentPage: number;
+  /** Number of the last page */
+  lastPage: number;
+  /** Returns the selected page number */
+  onChange: (page: number) => void;
+
+  /** Ref is forwarded to root element. Alternative for React `ref` attribute. */
+  forwardedRef?: React.RefObject<HTMLElement>;
+  /** Next page button label for screen readers  */
+  ariaNextButtonLabel: string;
+  /** Previous page button label for screen readers */
+  ariaPreviousButtonLabel: string;
+
+  pageInput?: boolean;
+
+  paginationInputProps?: PaginationInputProps;
+}
+
+// export type PaginationProps = inputProps & InternalPaginationProps;
+export type PaginationProps = InternalPaginationProps;
+
+const baseClassName = 'fi-pagination';
+
+export const paginationClassNames = {
+  styleWrapper: `${baseClassName}_style-wrapper`,
+  content: `${baseClassName}_content`,
+  contentWrapper: `${baseClassName}_contentWrapper`,
+  heading: `${baseClassName}_heading`,
+  textContentWrapper: `${baseClassName}_text-content-wrapper`,
+  icon: `${baseClassName}_icon`,
+  iconWrapper: `${baseClassName}_icon-wrapper`,
+  pageButton: `${baseClassName}_page-button`,
+  stockButton: `${baseClassName}_stock-button`,
+  smallScreen: `${baseClassName}--small-screen`,
+  pageInputWrapper: `${baseClassName}_page-input-wrapper`,
+  buttonsWrapper: `${baseClassName}_buttons-wrapper`,
+  actionElementWrapper: `${baseClassName}_action-element-wrapper`,
+  pageNumbers: `${baseClassName}_page-numbers`,
+};
+
+class BasePagination extends Component<PaginationProps> {
+  private onLeftButtonClick = () => {
+    this.props.onChange(this.props.currentPage - 1);
+  };
+
+  private onRightButtonClick = () => {
+    this.props.onChange(this.props.currentPage + 1);
+  };
+
+  private onNumberInputChange = (page: number) => {
+    this.props.onChange(page);
+  };
+
+  render() {
+    const {
+      children,
+      className,
+      asProp,
+      textFunction,
+      onChange,
+      currentPage,
+      lastPage,
+      ariaPreviousButtonLabel,
+      ariaNextButtonLabel,
+      pageInput,
+      paginationInputProps,
+      smallScreen,
+      ...passProps
+    } = this.props;
+    return (
+      <HtmlNav
+        {...passProps}
+        className={classnames(baseClassName, className, externalClassName, {
+          [paginationClassNames.smallScreen]: !!smallScreen,
+        })}
+      >
+        <HtmlDiv className={paginationClassNames.styleWrapper}>
+          <HtmlDiv className={paginationClassNames.buttonsWrapper}>
+            <Button
+              className={paginationClassNames.stockButton}
```

Comment:
Where does the `stockButton` name come from? Would `arrowButton` or something of the sort be clearer?

Reply:
`stockButton` was a remnant from earlier experiment. Renamed now.